src/org/labkey/cds/data/steps/PopulateStudiesTask.java

```diff
@@ -135,6 +151,99 @@ private void cleanContainers(Set<String> importStudies, Container project, User
         logger.info("Deleted " + deleted + " studies in " + DateUtil.formatDuration(finish - start) + ".");
     }
 
+    private void ensurePermissions(Logger logger) throws PipelineJobException
+    {
+        QuerySchema schema = DefaultSchema.get(user, project).getSchema("cds");
+
+        if (null == schema)
+            throw new PipelineJobException("Unable to find cds schema.");
+
+        TableInfo importStudyGroups = schema.getTable("import_studygroups");
+
+        if (null == importStudyGroups)
+            throw new PipelineJobException("Unable to find cds.import_studygroups table.");
+
+        SQLFragment sql = new SQLFragment("SELECT * FROM ").append(importStudyGroups);
+        Map<String, Object>[] importStudies = new SqlSelector(importStudyGroups.getSchema(), sql).getMapArray();
+
+        // Map((Study, Role) -> Set(Groups))
+        Map<Pair<String, String>, Set<String>> studyRoleGroups = new HashMap<>();
```

Comment:
Using Map<String,Map<Role, List<UserPrincipal>>> studyRoleGroups might be easier as policy is per study.